### PR TITLE
Issue 656

### PR DESCRIPTION
**Pull request by fmarmar**
Issue #656

### CODE REVIEW

File: src/main/java/net/masterthought/cucumber/json/Hook.java [hunk 1]
```diff
@@ -1,5 +1,9 @@
 package net.masterthought.cucumber.json;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/json/Hook.java]: First of all - commit message is not valid - check other commits for reference

File: src/main/resources/templates/macros/json/hooks.vm [hunk 2]
```diff
@@ -22,6 +22,7 @@
             #includeMessage("Error message", $hook.getResult().getErrorMessage())
           #end
 
+		  #includeOutput($hook.getOutputs(), $status.isPassed())
```
Comment (damianszczepanik) [inline src/main/resources/templates/macros/json/hooks.vm]: problem with formatting/indention ?

File: src/test/java/net/masterthought/cucumber/ReportGenerator.java [hunk 3]
```diff
@@ -23,6 +23,8 @@
     protected static final String EMPTY_JSON = "empty.json";
     protected static final String INVALID_JSON = "invalid.json";
     protected static final String INVALID_REPORT_JSON = "invalid-report.json";
+    
+    protected static final String SAMPLE_HOOKS_JSON = "sample-hooks.json";
```
Comment (damianszczepanik) [inline src/test/java/net/masterthought/cucumber/ReportGenerator.java]: I don't want to extra test file. Instead:
- add single output and array aoutput into existing sample.json file
- add integration test that validates if the output is generated properly

File: src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java [hunk 4]
```diff
@@ -332,6 +332,34 @@ public void generatePage_OnRubyFormat_ForAfterHook_generatesEmbedding() {
         embeddingsElement[0].hasImageContent(embeddings[0]);
         asserEmbeddingFileExist(embeddings[0]);
     }
+    
+    @Test
+    public void generatePage_ForHook_generatesOutput() {
```
Comment (damianszczepanik) [inline src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java]: as this feature is required for Ruby (and probably only for Ruby) refer to the naming convention of above method and apply

File: src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java [hunk 5]
```diff
@@ -332,6 +332,34 @@ public void generatePage_OnRubyFormat_ForAfterHook_generatesEmbedding() {
...
+        page.generatePage();
+
+        // then
+        DocumentAssertion document = documentFrom(page.getWebPage());
+        
+        Output[] beforeOutputElements = feature.getElements()[0].getBefore()[0].getOutputs();
+        OutputAssertion beforeOutput = document.getFeature().getElements()[0].getBefore().getHooks()[0].getOutput();
+        assertOutuput(beforeOutput, beforeOutputElements);
+
+        Output[] afterOutputElements = feature.getElements()[0].getAfter()[0].getOutputs();
```
Comment (damianszczepanik) [inline src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java]: Testing before OR after is enough - no need to test both

File: src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java [hunk 6]
```diff
@@ -332,6 +332,34 @@ public void generatePage_OnRubyFormat_ForAfterHook_generatesEmbedding() {
...
+        OutputAssertion afterOutput = document.getFeature().getElements()[0].getAfter().getHooks()[0].getOutput();
+        assertOutuput(afterOutput, afterOutputElements);
+      
+    }
+    
+    private void assertOutuput(OutputAssertion outputAssertion, Output[] outputElements) {
```
Comment (damianszczepanik) [inline src/test/java/net/masterthought/cucumber/generators/integrations/FeatureReportPageIntegrationTest.java]: merge, change to inline

File: src/test/resources/json/sample-hooks.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
+[
```
Comment (damianszczepanik) [inline src/test/resources/json/sample-hooks.json]: to delete

**Comment by damianszczepanik** [comment]
@fmarmar are you going to complete this PR?

**Comment by fmarmar** [comment]
@damianszczepanik I plan to but don't have the time right now. I'll try to work on it next weekend.

**Comment by pavelpp** [comment]
@fmarmar any hope we'll see a fix? It's already November...

**Comment by damianszczepanik** [comment]
Resolved as 'done'

**Pull request closed without merging**